web/rootfs/defaults/settings-config.js

```diff
@@ -0,0 +1,259 @@
+{{ $CONFIG_TESTING_CAP_SCREENSHARE_BITRATE := .Env.CONFIG_TESTING_CAP_SCREENSHARE_BITRATE | default "1" -}}
+{{ $ENABLE_LIPSYNC := .Env.ENABLE_LIPSYNC | default "false" | toBool -}}
+{{ $ENABLE_SUBDOMAINS := .Env.ENABLE_SUBDOMAINS | default "false" | toBool -}}
+{{ $ENABLE_TRANSCRIPTIONS := .Env.ENABLE_TRANSCRIPTIONS | default "false" | toBool -}}
+{{ $CONFIG_ENABLE_STATS_ID := .Env.CONFIG_ENABLE_STATS_ID | default "false" | toBool -}}
+{{ $CONFIG_OPEN_BRIDGE_CHANNEL := .Env.CONFIG_OPEN_BRIDGE_CHANNEL | default "datachannel" -}}
+{{ $CONFIG_DISABLE_BRIDGE_CHANNEL := .Env.CONFIG_DISABLE_BRIDGE_CHANNEL | default "false" | toBool -}}
+{{ $CONFIG_STEREO := .Env.CONFIG_STEREO | default "false" | toBool -}}
+{{ $CONFIG_ENABLE_TALK_WHILE_MUTED := .Env.CONFIG_ENABLE_TALK_WHILE_MUTED | default "false" | toBool -}}
+{{ $CONFIG_ENABLE_NO_AUDIO_DETECTION := .Env.CONFIG_ENABLE_NO_AUDIO_DETECTION | default "false" | toBool -}}
+{{ $CONFIG_ENABLE_CALENDAR := .Env.CONFIG_ENABLE_CALENDAR | default "false" | toBool -}}
+{{ $CONFIG_REQUIRE_DISPLAY_NAME := .Env.CONFIG_REQUIRE_DISPLAY_NAME | default "false" | toBool -}}
+{{ $CONFIG_REQUIRE_DISPLAY_NAME := .Env.CONFIG_REQUIRE_DISPLAY_NAME | default "false" | toBool -}}
+{{ $CONFIG_START_VIDEO_MUTED := .Env.CONFIG_START_VIDEO_MUTED | default 10 -}}
+{{ $CONFIG_START_AUDIO_MUTED := .Env.CONFIG_START_AUDIO_MUTED | default 10 -}}
+{{ $CONFIG_USE_STUN_TURN := .Env.CONFIG_USE_STUN_TURN | default "true" | toBool -}}
+{{ $CONFIG_USE_IPV6 := .Env.CONFIG_USE_IPV6 | default "true" | toBool -}}
+{{ $CONFIG_DEPLOYMENTINFO_USERREGION := .Env.CONFIG_DEPLOYMENTINFO_USERREGION | default "<!--# echo var=\"http_x_proxy_region\" default=\"\" -->" -}}
+
+{{ if $CONFIG_EXTERNAL_CONNECT -}}
+    {{ if $ENABLE_SUBDOMAINS -}}
+config.externalConnectUrl = '//{{ if .Env.CONFIG_BOSH_HOST }}{{ .Env.CONFIG_BOSH_HOST }}{{ end }}/<!--# echo var="subdir" default="" -->http-pre-bind';
+    {{ else -}}
+config.externalConnectUrl = '//{{ if .Env.CONFIG_BOSH_HOST }}{{ .Env.CONFIG_BOSH_HOST }}{{ end }}/http-pre-bind';
+    {{ end -}}
+{{ end -}}
+
+config.startAudioMuted = {{ $CONFIG_START_AUDIO_MUTED }};
+config.startVideoMuted = {{ $CONFIG_START_VIDEO_MUTED }};
+
+config.resolution = {{ $CONFIG_RESOLUTION }};
+if (!config.hasOwnProperty('constraints')) config.constraints={};
+if (!config.constraints.hasOwnProperty('video')) config.constraints.video={};
+config.constraints.video.height = {ideal: {{$CONFIG_RESOLUTION}}, max: {{$CONFIG_RESOLUTION}}, min: {{$CONFIG_RESOLUTION_MIN}}};
+config.constraints.video.width = {ideal: {{$CONFIG_RESOLUTION_WIDTH}}, max: {{$CONFIG_RESOLUTION_WIDTH}}, min: {{$CONFIG_RESOLUTION_WIDTH_MIN}}};
+
+config.enableTcc = {{ $CONFIG_ENABLE_TCC }};
+config.enableRemb = {{ $CONFIG_ENABLE_REMB }};
+
+// Enable / disable simulcast support.
+config.disableSimulcast = {{ $CONFIG_DISABLE_SIMULCAST }};
+
+// testing values, defaults should be sane
+if (!config.hasOwnProperty('testing')) config.testing={};
```

Comment:
```suggestion
Object.assign(config.testing, {});
```

I need to check if Duktape supports this though.

Reply:
Great, I've applied this everywhere

Reply:
Reverting this one:
Uncaught TypeError: Cannot convert undefined or null to object